src/MultiplayerMod/Multiplayer/World/WorldManager.cs

```diff
@@ -31,23 +31,28 @@ public class WorldManager {
         multiplayer.Players.ForEach(it => server.Send(new ChangePlayerStateCommand(it.Id, PlayerState.Loading)));
         server.Send(new ChangePlayerStateCommand(multiplayer.Players.Current.Id, PlayerState.Ready));
         server.Send(new ShowLoadOverlay());
-        server.Send(new LoadWorld(GetWorldSave()), MultiplayerCommandOptions.SkipHost);
+        server.Send(new LoadWorld(WorldName, WorldSave), MultiplayerCommandOptions.SkipHost);
         if (resume)
             server.Send(new ResumeGame());
     }
 
-    public static void LoadWorldSave(byte[] data) {
-        var path = Path.GetTempFileName();
+    public static void LoadWorldSave(string worldName, byte[] data) {
+        var path = Path.Combine(SaveLoader.GetSavePrefixAndCreateFolder(), worldName, $"{worldName}.sav");
+        Directory.CreateDirectory(Path.GetDirectoryName(path)!);
         using (var writer = new BinaryWriter(File.OpenWrite(path))) {
             writer.Write(data);
         }
         LoadScreen.DoLoad(path);
     }
 
-    private static byte[] GetWorldSave() {
-        var path = SaveLoader.GetActiveSaveFilePath();
-        Execution.RunUsingLevel(ExecutionLevel.Multiplayer, () => SaveLoader.Instance.Save(path));
-        return File.ReadAllBytes(path);
+    private static string WorldName => Path.GetFileNameWithoutExtension(SaveLoader.GetActiveSaveFilePath());
+
+    private static byte[] WorldSave {
```

Comment:
Are you sure it's a good idea to make it a property?🤔
I would expect it's cached or something. Maybe it's better to create a class, say `WorldSave` with `Name` and `Data` and just pass it to the command?

Reply:
I don't mind to revert it back.

In general I'm doubting any parametrless method starts with Get in C#. Those feels to be either properties or should be renamed to something else (e.g. Calculate...)